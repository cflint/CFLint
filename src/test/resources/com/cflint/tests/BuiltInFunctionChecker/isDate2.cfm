component{
    function foo(d){
        if( d && (isDate(1.5))){
            return 123;
        };
        return 321;
    }
}
