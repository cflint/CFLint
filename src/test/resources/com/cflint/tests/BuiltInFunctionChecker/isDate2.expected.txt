{
  "version" : "",
  "timestamp" : 1526095019,
  "issues" : [ {
    "severity" : "WARNING",
    "id" : "AVOID_USING_ISDATE",
    "message" : "AVOID_USING_ISDATE",
    "category" : "CFLINT",
    "abbrev" : "AU",
    "locations" : [ {
      "file" : "src\\test\\resources\\com\\cflint\\tests\\BuiltInFunctionChecker\\isDate2.cfm",
      "fileName" : "isDate2.cfm",
      "function" : "foo",
      "offset" : 50,
      "column" : 19,
      "line" : 3,
      "message" : "Avoid using the isDate built-in function.  It is too permissive.  Use isValid() instead.",
      "variable" : "",
      "expression" : "isDate(1.5)"
    } ]
  } ],
  "counts" : {
    "totalFiles" : 0,
    "totalLines" : 0,
    "countByCode" : [ {
      "code" : "AVOID_USING_ISDATE",
      "count" : 1
    } ],
    "countBySeverity" : [ {
      "severity" : "WARNING",
      "count" : 1
    } ]
  }
}